### ISSUE TITLE
[naming-convention] Precedence of property selector has been changed

### ISSUE THREAD

**Issue by susisu** [typescript-eslint/typescript-eslint#2860]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "memberLike",
        "format": ["camelCase"]
      },
      {
        "selector": "property",
        "format": ["camelCase", "PascalCase"]
      }
    ]
  }
}
```

```TS
const foo = { MyProperty: 42 };
```

**Expected Result**
Because typescript-eslint 4.8.2 doesn't report error for this, I expect typescript-eslint 4.9.x also doesn't report error.

**Actual Result**
The following error is reported.
```
  1:15  error  Object Literal Property name `MyProperty` must match one of the following formats: camelCase  @typescript-eslint/naming-convention
```

Previously `property` had a higher precedence than `memberLike`, but it is now a meta selector and the precedence seems to be lower (ref: https://github.com/typescript-eslint/typescript-eslint/pull/2807).

I know it works as expected if I use `"selector": ["classProperty", "objectLiteralProperty", "typeProperty"]` instead of `"selector": "property"`, but I report this issue because it feels like a breaking change in a minor upgrade for me.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.9.1` |
| `@typescript-eslint/parser`        | `4.9.1` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.15.0` |
| `node`                             | `14.6.0` |

**Comment by bradzacher** [comment]
yeah this shouldn't have changed.
It's an interesting outcome of how the numbers sum up.

By converting `property` and `method` to group selectors, they gained an equal precedence to `memberLike`, meaning they're ordered by numerical value.

By definition, `memberLike` is a `|` of those two selector's values (and some others), so `memberLike` is always going to have a larger numerical value as ofc `(property | method) > property`.

So we'll need to manually inflate both `property` and `method`'s enum value separately from `memberLike` so that they can be numerically larger than it - but it must be done in such a way that the bit field doesn't overlap with an existing selector.

E.g. something like
```ts
property = property | (Max(Selectors) << 1);
```
